common/oatbox/log/LoggerService.php

```diff
@@ -29,98 +30,104 @@ class LoggerService extends ConfigurableService implements LoggerInterface
 {
     use LoggerTrait;
 
-    const SERVICE_ID = 'generis/log';
+    public const SERVICE_ID = 'generis/log';
+    public const DEFAULT_CHANNEL = 'tao';
+    private const OPTION_LOGGER = 'logger';
+    private const OPTION_LOGGERS = 'loggers';
+    private const OPTION_CLASS = 'class';
+    private const OPTION_OPTIONS = 'options';
+    private const OPTION_NAME = 'name';
 
-    const LOGGER_OPTION = 'logger';
-
-    /** @var LoggerInterface */
-    protected $logger;
+    /** @var LoggerInterface[] */
+    private $loggers = [];
 
     /**
-     * Add a Psr3 logger to LoggerService instance
-     * Previous and new logger are encapsulated into a LoggerAggregator
-     *
-     * @param LoggerInterface $logger
-     * @return LoggerAggregator|LoggerInterface
+     * Register the given PSR-3 logger into the defined channel.
+     * Previous and new logger are encapsulated into a LoggerAggregator.
      */
-    public function addLogger(LoggerInterface $logger)
+    public function addLogger(LoggerInterface $logger, string $channel = null)
     {
-        if ($this->isLoggerLoaded() && !($this->logger instanceof NullLogger)) {
-            $logger = new LoggerAggregator([$logger, $this->logger]);
+        $channel = $channel ?? self::DEFAULT_CHANNEL;
+
+        if (!array_key_exists($channel, $this->loggers)) {
+            $this->loggers[$channel] = new NullLogger();
         }
-        $this->logger = $logger;
-        return $logger;
+
+        $this->loggers[$channel] = new LoggerAggregator([$logger, $this->loggers[$channel]]);
+
+        return $this->loggers[$channel];
     }
 
-    /**
-     * Get the logger
-     *
-     * @return LoggerInterface
-     */
-    public function getLogger()
+    public function getLogger(string $channel = null): LoggerInterface
     {
-        if (!$this->isLoggerLoaded()) {
-            $this->loadLogger();
+        $channel = $channel ?? self::DEFAULT_CHANNEL;
+
+        if ($this->loggers === []) {
+            // To keep backward compatibility, "logger" key must be supported also
+            if ($this->hasOption(self::OPTION_LOGGER)) {
+                $this->loadLogger($this->getOption(self::OPTION_LOGGER));
+            }
+
+            $this->loadLoggers();
+        }
+
+        if (array_key_exists($channel, $this->loggers) && $this->loggers[$channel] instanceof LoggerInterface) {
```

Comment:
you can remove multiple `$this->loggers[$channel]` access by key (3 times) by smth like:
```
return ($logger = $this->loggers[$channel] ?? null) && $logger instanceof LoggerInterface
    ? $logger
    : new NullLogger();
```